src/iexamine.cpp

```diff
@@ -1911,15 +1911,17 @@ void iexamine::plant_seed( player &p, const tripoint &examp, const itype_id &see
     } else {
         used_seed = p.use_amount( seed_id, 1 );
     }
-    used_seed.front().set_age( 0_turns );
-    g->m.add_item_or_charges( examp, used_seed.front() );
-    if( g->m.has_flag_furn( "PLANTABLE", examp ) ) {
-        g->m.furn_set( examp, furn_str_id( g->m.furn( examp )->plant->transform ) );
-    } else {
-        g->m.set( examp, t_dirt, f_plant_seed );
+    if( !used_seed.empty() ) {
+        used_seed.front().set_age( 0_turns );
+        g->m.add_item_or_charges( examp, used_seed.front() );
+        if( g->m.has_flag_furn( "PLANTABLE", examp ) ) {
+            g->m.furn_set( examp, furn_str_id( g->m.furn( examp )->plant->transform ) );
+        } else {
+            g->m.set( examp, t_dirt, f_plant_seed );
+        }
+        p.moves -= to_moves<int>( 30_seconds );
+        add_msg( _( "%s planted %s." ), p.disp_name(), item::nname( seed_id ) );
```

Comment:
```suggestion
        p.add_msg_player_or_npc( _( "You plant some %s." ), _("<npcname> plants some %s."), item::nname( seed_id ) );
```
`add_msg_player_or_npc` also includes a check for the NPC being visible to the player character, so it only prints the message when you can actually observe it.